### PR TITLE
Fix #3332: fixed varbyte & time issue

### LINKED ISSUE #3332: Ingestion - Redshift varbyte & time
**Affected module**
Ingestion

**Describe the bug**
```
/Users/pmbrull/venv/lib/python3.9/site-packages/metadata/ingestion/source/redshift.py:120: SAWarning: Did not recognize type 'binary varying' of column 'my_varbyte'
  column_info = super(RedshiftDialectMixin, self)._get_column_info(
```

When ingesting redshift data of type `varbinary` we get the error above and the column gets mapped wrongly to varchar.

Also, although there is no error in the logs, the `TIME` type gets ingested as `varchar` as well.

**To Reproduce**

Reproduce this by running the ingestion against our DEV redshift and check the table `public.type_test`.

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

Fix #3332: fixed varbyte & time issue

#
### Type of change :

- [x] Bug fix
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @ayush-shah @pmbrull

### CODE REVIEW

**Review by ayush-shah — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/redshift.py [hunk 1]
```diff
@@ -208,6 +210,176 @@ def _get_schema_column_info(self, connection, schema=None, **kw):
...
+    default,
+    notnull,
+    domains,
+    enums,
+    schema,
+    comment,
+    generated,
+    identity,
+):
+    def _handle_array_type(attype):
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/redshift.py:227]: Why are we defining this function inside the body?

**Review by pmbrull — CHANGES_REQUESTED** [review]
Thanks, @ulixius9, for the PR. One comment here would be that the function feels super complicated. I've seen that when handling a lot of the internal logic in the ingestion module we end up having big nested logics that are hard to understand, which makes them hard to maintain and debug.

Could we try to break this process down into smaller, more manageable and unit-tested functions? Let's try to spend a bit more time on improving the codebase in terms of overall readability and testing.

If this is urgent to deliver for some users, please feel free to merge it, but then open a separate issue to try to improve this.

Thanks!

**Review by pmbrull — COMMENTED** [review]
Thanks for the changes! If possible, please add some unit tests on the individual functions, which will help us evolve the code and showcase what is going on 🙏

**Review by pmbrull — APPROVED** [review]
As with the other PR, let's proceed with pending further work on unit testing. Once SonarCloud correctly collects data, let's iterate over all connectors. Thanks!

**Pull request merged**